src/auth/googleauth.ts

```diff
@@ -439,10 +445,11 @@ export class GoogleAuth {
         json as ExternalAccountClientOptions,
         options
       )!;
-      client.scopes = this.scopes;
+      client.scopes = this.scopes || this.defaultScopes;
```

Comment:
do you think we do this enough now that we should consider pulling this into a private helper, like:

```js
private getAnyScopes() {
  return this.scopes || this.defaultScopes
}
```

trying to decide if that makes us less likely to forget the `|| this.defaultScopes` as we refactor in the future.

Reply:
Good idea, I created the helper function. We use this in 5 places. I also think this makes it less likely to forget the `|| this.defaultScopes` by using the helper function.